src/dotnet-interactive-vscode/insiders/src/notebookControllers.ts

```diff
@@ -118,28 +118,25 @@ export class DotNetNotebookKernel {
         const executionTask = controller.createNotebookCellExecution(cell);
         if (executionTask) {
             executionTasks.set(cell.document.uri.toString(), executionTask);
+            let outputUpdatePromise = Promise.resolve();
             try {
                 const startTime = Date.now();
                 executionTask.start(startTime);
                 executionTask.clearOutput(cell);
                 const controllerErrors: vscodeLike.NotebookCellOutput[] = [];
 
                 function outputObserver(outputs: Array<vscodeLike.NotebookCellOutput>) {
-                    updateCellOutputs(executionTask!, [...outputs, ...controllerErrors]).then(() => { });
+                    outputUpdatePromise = outputUpdatePromise.catch(ex => {
```

Comment:
I don't see `outputUpdatePromise` ever being assigned any work other than the pre-resolved promise which means the `catch()` block will never get hit and it really reduces to just the `finally()` call; let's collapse this to be similar to the original behavor, but with this new `catch()` block attached, e.g.,

``` typescript
outputUpdatePromise = updateCellOutputs(...).catch(ex => {
    console.error(...);
});
```

(or whatever makes the most sense)

Reply:
There's a little bit more to the way its used. We update the promise everytime, so as to chain the promises. basically the subsequent update can only be done after the previous update has been completed (this is achieved by chaining the promises).